docs/datagen/client/modelproviders.md

```diff
@@ -393,7 +393,7 @@ Once the data provider is running, the models within the `ModelProvider` subclas
 @Override
 public CompletableFuture<?> run(CachedOutput cache) {
   // Populate the model provider
-  this.example.generateAll(cache); // Generate the models
+  CompletableFuture<?> exampleFutures = this.example.generateAll(cache); // Generate the models
```

Comment:
Should this show combining the given `CompletableFuture` with the one from its own `generateAll`? As in, showing the use of `CompletableFuture.allOf` (or another suitable alternative) wherein all CFs from calls to `generateAll`s are passed in as arguments?

Reply:
Sure, that's fine.